packages/clerk-js/src/ui/utils/useFormControl.ts

```diff
@@ -60,13 +66,17 @@ export const useFormControl = <Id extends string>(
     placeholder: '',
     options: [],
     enableErrorAfterBlur: false,
+    direction: '',
   };
+
   const { translateError } = useLocalizations();
   const [value, setValueInternal] = React.useState<string>(initialState);
   const [checked, setCheckedInternal] = React.useState<boolean>(opts?.checked || false);
   const [errorText, setErrorText] = React.useState<string | undefined>(undefined);
+  const [warningText, setWarningText] = React.useState<string | undefined>(undefined);
```

Comment:
🔧 It would be simpler to set the default value of the error and warning texts to `""` and avoid the <string | undefined> type. That would also simplify the following `typeof` checks as you can just do `if (!warningText)`.